app/src/main/java/org/mozilla/focus/navigation/TransactionHelper.java

```diff
@@ -101,30 +94,20 @@ boolean shouldFinish() {
         return EntryData.TYPE_ROOT == getEntryType(lastEntry);
     }
 
-    void updateForegroundType(@EntryData.EntryType int type) {
-        FragmentManager manager = activity.getSupportFragmentManager();
-        int size = manager.getBackStackEntryCount();
-        if (size == 0) {
-            return;
-        }
-        FragmentManager.BackStackEntry entry = manager.getBackStackEntryAt(size - 1);
-        if (this.entryDataMap.get(entry).type != type) {
-            this.entryDataMap.get(entry).type = type;
-        }
-    }
-
     void popAllScreens() {
-        popScreensUntil(null);
+        popScreensUntil(null, EntryData.TYPE_ROOT);
     }
 
-    boolean popScreensUntil(@Nullable String targetEntryName) {
+    boolean popScreensUntil(@Nullable String targetEntryName, @EntryData.EntryType int type) {
```

Comment:
split into popScreensUntil(String) and popScreensUntil(String, int) to improve readability.